File: Day29_AddSum.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
    head = ListNode()
    temp = head
    c = 0
    while(l1 or l2 or c):
        if(l1):
            n1 = l1.val
            l1 = l1.next
        else:
            n1 = 0
        if(l2):
            n2 = l2.val
            l2 = l2.next
        else:
            n2 = 0
        r = n1 + n2 + c
        c = r // 10
        r = r % 10
        newn = ListNode(r)
        temp.next = newn
        temp = newn
    return head.next
        

File: Day29_AddSum.cpp
#include <bits/stdc++.h>
using namespace std;
ifstream in("in.txt");
ofstream out("out.txt");

ListNode *addTwoNumbers(ListNode *l1, ListNode *l2)
{
    ListNode *head = new ListNode();
    ListNode *temp = head;
    int c = 0;
    while (l1 or l2 or c)
    {
        int n1, n2;
        if (l1)
        {
            n1 = l1->val;
            l1 = l1->next;
        }
        else
            n1 = 0;
        if (l2)
        {
            n2 = l2->val;
            l2 = l2->next;
        }
        else
            n2 = 0;
        int r = n1 + n2 + c;
        c = r / 10;
        r = r % 10;
        ListNode *newn = new ListNode();
        newn->val = r;
        newn->next = nullptr;
        temp->next = newn;
        temp = newn;
    }
    return head->next;
}

int main()
{

    return 0;
}